File: go/usermanual/PublicSysResources.go
package usermanual

import (
	"github.com/lvcdy/huawei-lte-api-go/session"
)

// PublicSysResources 对应 usermanual/PublicSysResources.py。
type PublicSysResources struct {
	*session.ApiGroup
}

// NewPublicSysResources 创建 PublicSysResources API 分组。
func NewPublicSysResources(s *session.Session) *PublicSysResources {
	return &PublicSysResources{ApiGroup: session.NewApiGroup(s)}
}

// Config 对应 public_sys-resources/config.xml。
func (p *PublicSysResources) Config() (map[string]interface{}, error) {
	return p.S.Get("public_sys-resources/config.xml", nil, "usermanual")
}

File: src/usermanual/public_sys_resources.rs
//! Public Sys Resources usermanual group (`usermanual/PublicSysResources.py`).

use serde_json::Value;

use crate::api::get_value;
use crate::connection::Connection;
use crate::errors::Result;

/// PublicSysResources usermanual group.
pub struct PublicSysResources<'a> {
    conn: &'a Connection,
}

impl<'a> PublicSysResources<'a> {
    /// Build a group bound to `conn`.
    pub fn new(conn: &'a Connection) -> Self {
        PublicSysResources { conn }
    }

    /// `public_sys-resources/config.xml`. Public system resources configuration.
    pub fn config(&self) -> Result<Value> {
        get_value(self.conn, "usermanual/public_sys-resources/config.xml")
    }
}
